### DISCUSSION TITLE
How can I `go get` tsh packages, to embed `tsh` in a similar way to how `tsh` embeds `kubectl`?

### DISCUSSION THREAD

**Discussion by simoncrowe** [gravitational/teleport#40955] [Q&A]
I'm part of a platform team working on a Go-based CLI to provide access to Kubernetes clusters, databases, test environments and so on.

We want to add shortcuts to perform common operations that don't exist in `tsh`. Internally, we use a python-based CLI that wraps `tsh` with subprocess calls. One downside of this is that it's difficult to package and distribute to people working on different operating systems.

In an ideal word, we would import what we need from `tsh`, add our functionality and compile the result into a static binary and ship this to users of the platform.

Here's an example of how `kubectl` code is imported into `tsh`: https://github.com/gravitational/teleport/blob/v14.3.16/tool/tsh/common/kube.go#L49-L53

For Teleport version 14, if I run
```shell
go get github.com/gravitational/teleport/tool/tsh/common@v0.0.0-20240327115209-ecc37f7d89ab
```
I eventually get this error:
```
go: github.com/gravitational/teleport/tool/tsh/common imports
	github.com/gravitational/teleport/lib/benchmark imports
	github.com/gravitational/teleport/lib/web imports
	github.com/gravitational/teleport/lib/reversetunnelclient imports
	github.com/gravitational/teleport/lib/proxy/peer imports
	github.com/gravitational/teleport/api/utils/grpc/stream: reading github.com/gravitational/teleport/api/go.mod at revision api/v0.0.0: unknown revision api/v0.0.0
```

I'm not sure if I'm missing something because I'm unfamiliar with how go packaging works, of if version 14 of the code at least, cannot be imported into a n external Go module.

**Comment by Tener** [comment]
You may want to consider using `tsh aliases`: https://goteleport.com/docs/connect-your-client/tsh/#custom-aliases-and-defaults

Depending on your exact use case these may work for you.
